### PR TITLE
VBM-185: email verification after sign up added

### PR DESCRIPTION
**Pull request by VladimirBludov**

### CODE REVIEW

File: packages/mobile/src/common/types/verify/verify.ts [hunk 1]
```diff
@@ -2,4 +2,20 @@ type VerifyPhoneRequestDto = {
...
+  email: string;
+};
+
+type PropsVerifyScreens = {
+  route: {
+    params: {
+      fromSignUp?: boolean;
+    };
+  };
+};
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/common/types/verify/verify.ts:15]: You can use navigation types for this, e.g.:
`type VerifyEmailScreenProps = NativeStackScreenProps<RootNavigationParamList, typeof RootScreenName.VERIFY_EMAIL>`

File: packages/mobile/src/screens/verify-screens/verified-email/verified-email.tsx [hunk 2]
```diff
@@ -1,29 +1,51 @@
...
 
 const VerifiedEmailScreen: FC = () => {
   const { t } = useTranslation();
   const navigation = useNavigation<RootNavigationProps>();
   const { colors } = useCustomTheme();
+  const user = useAppSelector(selectCurrentUser);
+  const dataStatusAuth = useAppSelector(selectAuthDataStatus);
+  const isLoading = dataStatusAuth === DataStatus.PENDING;
+  const verifiedTitle =
+    user?.emailVerified && user.phoneVerified
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/verify-screens/verified-email/verified-email.tsx:35]:
```suggestion
    user?.emailVerified && user?.phoneVerified
```

File: packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx [hunk 3]
```diff
@@ -51,12 +56,25 @@ const VerifyCodeEmailScreen: FC = () => {
...
   });
+  const [isCorrectCode, setIsCorrectCode] = useState(false);
+
+  useEffect(() => {
+    const subscription = watch((value) => {
+      if (value?.code?.match(VERIFICATION_CODE_REGEX)) {
+        return setIsCorrectCode(true);
+      }
+
+      return setIsCorrectCode(false);
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx:73]:
```suggestion
      setIsCorrectCode(value.code.match(VERIFICATION_CODE_REGEX));
      return;
```

File: packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx [hunk 4]
```diff
@@ -78,7 +96,9 @@ const VerifyCodeEmailScreen: FC = () => {
     dispatch(verifyActions.verifyEmail(payload))
       .unwrap()
       .then(() => {
-        navigation.navigate(RootScreenName.VERIFIED_EMAIL);
+        navigation.navigate(RootScreenName.VERIFIED_EMAIL, {
+          fromSignUp: fromSignUp || false,
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx:100]:
```suggestion
          fromSignUp: Boolean(fromSignUp),
```

File: packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx [hunk 5]
```diff
@@ -51,12 +56,25 @@ const VerifyCodePhoneScreen: FC = () => {
...
     defaultValues: {
       code: '',
     },
     validationSchema: codeSchema,
   });
+  const [isCorrectCode, setIsCorrectCode] = useState(false);
+
+  useEffect(() => {
+    const subscription = watch((value) => {
+      if (value?.code?.match(VERIFICATION_CODE_REGEX)) {
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx:69]: Same recommendation as above

File: packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx [hunk 6]
```diff
@@ -78,7 +96,9 @@ const VerifyCodePhoneScreen: FC = () => {
     dispatch(verifyActions.verifyPhone(payload))
       .unwrap()
       .then(() => {
-        navigation.navigate(RootScreenName.VERIFIED_PHONE);
+        navigation.navigate(RootScreenName.VERIFIED_PHONE, {
+          fromSignUp: fromSignUp || false,
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx:100]: ...and here

File: packages/mobile/src/store/auth/actions.ts [hunk 7]
```diff
@@ -51,8 +52,9 @@ const getCurrentUser = createAsyncThunk<UserDto, undefined, AsyncThunkConfig>(
 
 const logOut = createAsyncThunk<null, undefined, AsyncThunkConfig>(
   ActionType.LOG_OUT,
-  async (_, { extra }) => {
+  async (_, { extra, dispatch }) => {
     const { storage } = extra;
+    dispatch(personalInfoActions.resetPersonalInfo());
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/store/auth/actions.ts:57]: Wouldn't it be better to watch for logout action within personal info reducer?

**Review by AleksandrTermenzhy — APPROVED** [review]

File: packages/mobile/src/common/types/verify/verify.ts (hunk 1 above)
Comment (VladimirBludov) [reply packages/mobile/src/common/types/verify/verify.ts:15]: corrected

File: packages/mobile/src/screens/verify-screens/verified-email/verified-email.tsx (hunk 2 above)
Comment (VladimirBludov) [reply packages/mobile/src/screens/verify-screens/verified-email/verified-email.tsx:35]: corrected

File: packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx (hunk 3 above)
Comment (VladimirBludov) [reply packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx:73]: corrected

File: packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx (hunk 4 above)
Comment (VladimirBludov) [reply packages/mobile/src/screens/verify-screens/verify-code-email/verify-code-email.tsx:100]: corrected

File: packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx (hunk 6 above)
Comment (VladimirBludov) [reply packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx:100]: corrected

File: packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx (hunk 5 above)
Comment (VladimirBludov) [reply packages/mobile/src/screens/verify-screens/verify-code-phone/verify-code-phone.tsx:69]: corrected

**Review by mazurGit — APPROVED** [review]

**Review by Val-M9 — APPROVED** [review]

**Pull request merged**